{"cells":[{"cell_type":"code","execution_count":3,"metadata":{"executionInfo":{"elapsed":1213,"status":"ok","timestamp":1728939976104,"user":{"displayName":"Cyro Queiroz","userId":"10205326658499773298"},"user_tz":180},"id":"WNnSViHy-Qnz"},"outputs":[],"source":["import pandas as pd\n","import requests\n","import json\n","import re\n","import html\n"]},{"cell_type":"code","execution_count":4,"metadata":{"executionInfo":{"elapsed":483,"status":"ok","timestamp":1728939976582,"user":{"displayName":"Cyro Queiroz","userId":"10205326658499773298"},"user_tz":180},"id":"JCKVXIx0-Td4"},"outputs":[],"source":["# Define the API endpoint\n","#type = [week, industry]\n","#sessionID = \n","#abstractAPI = \n","#dayID = (from 2 until gets nothin)\n","#presentationID = \n","\n","#daysurl = https://ueg2024.planner.documedias.systems/api/program/rooms/2?program_type={$type}&system=program_week_2024&program_mode=list&program_sort=date&filter_display_type=1,3&filter_group=is_not_child\n","sessionurl = \"https://ueg2024.planner.documedias.systems/api/program/sessions/80?program_type=week&system=program_week_2024\"\n","#abstracturl = https://ueg2024.planner.documedias.systems/api/program/presentations/2506?program_type=week&system=program_week_2024\n","\n","# Make the API request\n","responseSession = requests.get(sessionurl)\n","dataSession = responseSession.json()\n","\n","# ITERATE OVER EACH FIELD"]},{"cell_type":"code","execution_count":8,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["Indication: ['Crohn', \"Crohn's disease\", 'IBD', 'inflammatory bowel disease', 'Ulcerative Colitis', 'Ulcerative', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '']\n","Asset: ['Tulisokibart', 'Ustekinumab', 'Guselkumab', 'Upadacitinib', 'Risankizumab', 'Mirikizumab', 'Ozanimod', 'Etrasimod', 'CT-P13', 'ABX-464', 'MORF-057', 'VTX958', 'TEV-48574', 'Humira', 'PF-07054894', 'PF-07261271', 'Ritlecitinib', 'PF-06480605', 'Infliximab', 'Cimzia', 'Golimumab', 'Vedolizumab', 'SHR0302', 'Neihulizumab', 'TAK-279', 'Tysabri', 'AGMB-129']\n","Company Name: ['Merck', 'JnJ', 'Prometheus', 'AbbVie', 'Eli Lilly', 'BMS', 'Pfizer', 'Celltrion', 'Abivax', 'Morphic', 'Ventyx', 'Sanofi', 'UCB', 'Takeda', 'Reistone', 'Altru Bio', 'Biogen', 'Agomab', 'Boehringer', 'Incannex', 'Medibiofarma', 'EA Pharma', 'Gilead', 'Spyre', 'Mozart', 'Athos', 'Syneos']\n","MoA: ['TL1A', 'Anti-IL-12/23', 'Interleukin 23 antagonist', 'JAK inhibitor', 'S1PR modulators', 'Anti TNF', 'MiRNA-124 expression stimulants', 'alpha4ß7 integrin', 'TYK2 inhibitor', 'CCR6 antagonist', 'Janus kinase 3 inhibitor', 'Janus kinase 1 inhibitor', 'Apoptosis stimulants', 'Alpha4beta1 integrin', 'Transforming growth factor beta receptor 1', 'Kinase inhibitor', 'Cannabinoid agonist', 'ROR-gamma-t agonist', 'PPARg Partial Agonist', 'aIL23p19 + aTNF', 'Restore CD8 T cell function', 'Targeting gene A', 'CD30 ligand', 'Pegylated human adrenomedullin', 'Targets T cells subset', 'TREM-1 antibody', 'Ion channel protein Kv1.3 inh']\n","Other Keywords: ['Pediatric', 'Paediatric', 'Intestinal ultrasound', 'CDST', 'Artificial Intelligence', 'Machine Learning', 'ultrasound', 'Clinical Decision support tool', 'Clinical decision', 'IUS', 'Combination', 'Meta-analysis', 'Comparative', 'Sequencing', 'Comparison', 'Fatigue', 'Adolescent', '', '', '', '', '', '', '', '', '', '']\n"]}],"source":["def clean_text(bad_text):\n","    # 1. Decode HTML entities (e.g., &amp;, &lt;, etc.)\n","    decoded_text = html.unescape(bad_text)\n","    \n","    # 2. Remove HTML tags using regex\n","    clean_text = re.sub(r'<.*?>{}', '', decoded_text)\n","    \n","    # 3. Strip any excessive whitespace\n","    clean_text = clean_text.strip()\n","    \n","    return clean_text\n","\n","def clean_dict(data_dict):\n","    cleaned_data = {}\n","    for key, value in data_dict.sessions():\n","        if isinstance(value, str):  # Only clean if it's a string\n","            cleaned_data[key] = clean_text(value)\n","        else:\n","            cleaned_data[key] = value  # Leave other types (lists, None, etc.) unchanged\n","    return cleaned_data\n","\n","# Assuming you've already loaded keywords_data from the JSON file\n","with open('keywords.json', 'r') as f:\n","    keywords_data = json.load(f)\n","\n","# Define the lists\n","indication = []\n","asset = []\n","company_name = []\n","MoA = []\n","other_keywords = []\n","\n","# Populate the lists from the keywords dictionary\n","for row in keywords_data['rows']:\n","    indication.append(row[0])\n","    asset.append(row[1])\n","    company_name.append(row[2])\n","    MoA.append(row[3])\n","    other_keywords.append(row[4])\n","\n","def find_matching_keywords(text, keywords):\n","    if not text:\n","        return None\n","    matches = [keyword for keyword in keywords if keyword and keyword.lower() in text.lower()]\n","    return '; '.join(matches) if matches else None\n","\n","# Print the lists to verify\n","print(\"Indication:\", indication)\n","print(\"Asset:\", asset)\n","print(\"Company Name:\", company_name)\n","print(\"MoA:\", MoA)\n","print(\"Other Keywords:\", other_keywords)\n"]},{"cell_type":"code","execution_count":14,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["<i><u>B.E. Sands</u><sup>1</sup>, G. D’Haens<sup>2</sup>, S. Danese<sup>3</sup>, T. Hisamatsu<sup>4</sup>, W. Reinisch<sup>5</sup>, N.A. Terry<sup>6</sup>, L. Salese<sup>6</sup>, R. Van Rampelbergh<sup>7</sup>, Z. Yang<sup>6</sup>, J. Johanns<sup>6</sup>, G. DuVall<sup>8</sup>, N. Abu Farsakh<sup>9</sup>, R. Panaccione<sup>10</sup></i><br>\n","<div><sup>1</sup>Dr. Henry D Janowitz Division of Gastroenterology, Icahn School of Medicine at Mount Sinai, New York, United States of America, <sup>2</sup>Department of Gastroenterology, Amsterdam University Medical Centers, Amsterdam, Netherlands, <sup>3</sup>Gastroenterology and Endoscopy, IRCCS San Raffaele Hospital and Vita-Salute San Raffaele University, Milan, Italy, <sup>4</sup>Department of Gastroenterology and Hepatology, Kyorin University, Tokyo, Japan, <sup>5</sup>Division of Gastroenterology & Hepatology, Medical University of Vienna, Vienna, Austria, <sup>6</sup>Janssen Research & Development, LLC, Spring House, United States of America, <sup>7</sup>Janssen Research & Development, Antwerp, Belgium, <sup>8</sup>Tyler Research Institute, LLC, Tyler, United States of America, <sup>9</sup>King Abdullah University Hospital, Irbid, Jordan, <sup>10</sup>Inflammatory Bowel Disease Unit, Division of Gastroenterology and Hepatology, University of Calgary, Calgary, Canada<br><br></div>\n","\n","            <div><strong>Introduction</strong>:&nbsp;GALAXI 2 &amp; 3 are identically designed 48-week, randomized, double-blind, double-dummy, placebo (PBO)- and active-comparator, treat-through registrational trials assessing the efficacy and safety of IV induction and SC maintenance therapy with guselkumab (GUS), a dual-acting IL-23p19 subunit inhibitor, in participants (pts) with moderately to severely active Crohn’s disease (CD). Primary results of each study were previously reported; here we explore the impact of biologic therapy exposure on efficacy outcomes with GUS compared to PBO in the pooled GALAXI 2 &amp; 3 dataset.</div>\n","                <div><strong>Aims & Methods</strong>:&nbsp;Pts with inadequate response or intolerance to conventional therapy or biologic therapy (BIO-ir), active CD (CDAI 220–450 + mean daily SF&gt;3 or AP&gt;1), and SES-CD≥6 (≥4 for isolated ileal disease) were randomly assigned 2:2:2:1 at week (Wk) 0 to GUS 200mg IV q4w (x3)→100mg SC q8w, GUS 200mg IV q4w (x3)→200mg SC q4w, UST IV→SC, or PBO. In each trial, the composite co-primary endpoints were 1) clinical response at Wk12 + clinical remission at Wk48, and 2) clinical response at Wk12 + endoscopic response at Wk48, comparing each GUS regimen to PBO. Major secondary endpoints included clinical remission at Wk12 and endoscopic response at Wk12. Analyses of BIO-ir and BIO-naïve subgroups in the individual trials were prespecified; pooled analyses were performed post hoc.</div>\n","                <div><strong>Results</strong>:&nbsp;​In the pooled dataset, 52% (378/730) of pts randomized to PBO or GUS had a prior history of BIO-ir and 42% (305/730) were BIO-naïve (6% [47/730] had prior exposure to biologics but no documented failure). At Wk12, GUS achieved higher rates of clinical remission and endoscopic response compared to PBO in the overall population and BIO-ir and BIO-naïve subgroups (<strong>Table</strong>). For the composite co-primary endpoints, both GUS regimens also achieved higher rates of 1) clinical response at Wk12 + clinical remission at Wk48 and 2) clinical response at Wk12 + endoscopic response at Wk48 compared to PBO in the overall population and BIO-ir and BIO-naïve subgroups.<br><br><table style=\"width: 100%; margin-left: calc(0%);\" class=\"\"><tbody><tr><td style=\"width: 100%;\" colspan=\"10\" class=\"fr-selected-cell\"><strong>Ta</strong><strong>ble. Efficacy of guselkumab (GUS) and placebo (PBO) for short- and long-term clinical and endoscopic endpoints in all participants and BIO-ir and BIO-naïve subgroups in the pooled GALAXI 2 &amp; 3 dataset</strong><br></td></tr><tr><td style=\"width: 10.0069%;\" class=\"\"><br></td><td style=\"width: 30%; text-align: center;\" colspan=\"3\"><strong>Overall population</strong><br></td><td style=\"width: 29.9523%; text-align: center;\" colspan=\"3\"><strong>Participants with a history of inadequate response or intolerance to at least 1 prior biologic<sup>a</sup> (BIO-ir)</strong><br></td><td style=\"width: 30%; text-align: center;\" colspan=\"3\"><strong>Participants with no history of exposure to biologics<sup>a</sup> (BIO-naïve) </strong><br></td></tr><tr><td style=\"width: 10.0068%; background-color: rgb(239, 239, 239);\" class=\"\"><br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=148<br></td><td style=\"width: 20%; background-color: rgb(239, 239, 239); text-align: center;\" colspan=\"2\"><strong>GUS 200mg IV combined</strong><sup><strong>b</strong></sup><br>N=582<br></td><td style=\"width: 10.0068%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=78<br></td><td style=\"width: 20%; background-color: rgb(239, 239, 239); text-align: center;\" colspan=\"2\"><strong>GUS 200mg IV combined</strong><sup><strong>b</strong></sup><br>N=300<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=61<br></td><td style=\"width: 20%; background-color: rgb(239, 239, 239); text-align: center;\" colspan=\"2\"><strong>GUS 200mg IV combined</strong><sup><strong>b</strong></sup><br>N=244<br></td></tr><tr><td style=\"width: 10.0068%;\"><strong>Clinical remission at Wk12</strong><br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\">28 (18.9%)<br></td><td style=\"width: 20%; text-align: center;\" colspan=\"2\">274 (47.1%)<br><strong>∆ 28.0%</strong><br>P&lt;.001*<br></td><td style=\"width: 10.0068%; text-align: center; vertical-align: middle;\" class=\"\">15 (19.2%)<br></td><td style=\"width: 20%; text-align: center;\" colspan=\"2\">138 (46.0%)<br><strong>∆ 26.9%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">10 (16.4%)<br></td><td style=\"width: 20%; text-align: center;\" colspan=\"2\">121 (49.6%)<br><strong>∆ 32.1%</strong><br>P&lt;.001*<br></td></tr><tr><td style=\"width: 10.0068%;\"><strong>Endoscopic response at Wk12</strong><br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">18 (12.2%)<br></td><td style=\"width: 20.0136%; text-align: center;\" colspan=\"2\">215 (36.9%)<br><strong>∆ 25.2%</strong><br>P&lt;.001*<br></td><td style=\"width: 10.0068%; text-align: center; vertical-align: middle;\" class=\"\">5 (6.4%)<br></td><td style=\"width: 20.0136%; text-align: center;\" colspan=\"2\">87 (29.0%)<br><strong>∆ 22.6%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">11 (18.0%)<br></td><td style=\"width: 20.0136%; text-align: center;\" colspan=\"2\">113 (46.3%)<br><strong>∆ 27.7%</strong><br>P&lt;.001*<br></td></tr><tr><td style=\"width: 10.0068%; background-color: rgb(239, 239, 239);\" class=\"\"><br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=148<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 100mg</strong><sup><strong>c</strong></sup><br>N=286<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 200mg</strong><sup><strong>d</strong></sup><br>N=296<br></td><td style=\"width: 10.0068%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=78<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 100mg</strong><sup><strong>c</strong></sup><br>N=153<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 200mg</strong><sup><strong>d</strong></sup><br>N=147<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=61<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 100mg</strong><sup><strong>c</strong></sup><br>N=116<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 200mg</strong><sup><strong>d</strong></sup><br>N=128<br></td></tr><tr><td style=\"width: 10.0068%;\"><strong>Clinical response at Wk12 + clinical remission at Wk48<br></strong>(composite co-primary endpoint)<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">18 (12.2%)<br></td><td style=\"width: 10%; text-align: center;\">137 (47.9%)<br><strong>∆ 35.8%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">152 (51.4%)<br><strong>∆ 39.0%</strong><br>P&lt;.001*<br></td><td style=\"width: 10.0068%; text-align: center; vertical-align: middle;\" class=\"\">10 (12.8%)<br></td><td style=\"width: 10%; text-align: center;\">70 (45.8%)<br><strong>∆ 33.2%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">73 (49.7%)<br><strong>∆ 37.4%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">7 (11.5%)<br></td><td style=\"width: 10%; text-align: center;\">60 (51.7%)<br><strong>∆ 40.4%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">70 (54.7%)<br><strong>∆ 41.8%</strong><br>P&lt;.001*<br></td></tr><tr><td style=\"width: 10.0068%;\"><strong>Clinical response at Wk12 + endoscopic response at Wk48</strong><br>(composite co-primary endpoint)<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">8 (5.4%)<br></td><td style=\"width: 10%; text-align: center;\">104 (36.4%)<br><strong>∆ 30.7%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">110 (37.2%)<br><strong>∆ 31.8%</strong><br>P&lt;.001*<br></td><td style=\"width: 10.0068%; text-align: center; vertical-align: middle;\" class=\"\">4 (5.1%)<br></td><td style=\"width: 10%; text-align: center;\">55 (35.9%)<br><strong>∆ 30.9%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">46 (31.3%)<br><strong>∆ 26.7%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">4 (6.6%)<br></td><td style=\"width: 10%; text-align: center;\">47 (40.5%)<br><strong>∆ 33.0%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">56 (43.8%)<br><strong>∆ 36.5%</strong><br>P&lt;.001*<br></td></tr><tr><td style=\"width: 100%;\" colspan=\"10\">Data are presented as n (%); ∆% (adjusted treatment difference) vs PBO; p-value vs PBO.<br>Adjusted treatment differences and p-values were based on the common risk difference by use of Mantel-Haenszel stratum weights and the Sato variance estimator. The stratification variables used are baseline CDAI score (≤300 or &gt;300), baseline SES-CD score (≤12 or &gt;12), BIO-ir status (Yes or No; this variable used only in analyses of the overall population), and baseline corticosteroid use (Yes or No). Population N values reflect the primary analysis set, defined as all randomized participants who received at least 1 (partial or complete) dose of study intervention and had a screening SES-CD score ≥6 (or ≥4 for participants with isolated ileal disease). PBO pts not in clinical response at week (Wk) 12 switched to UST.<br><br><u>Endpoint definitions</u>: clinical remission, CDAI &lt; 150; endoscopic response, ≥50% improvement from baseline in SES-CD or SES-CD ≤ 2; clinical response, ≥100-point reduction from baseline in CDAI or CDAI &lt; 150. Participants with treatment failure or missing data were considered to not have met the endpoint.<br><br>a. Anti-TNF agents or vedolizumab<br>b. Participants assigned to either GUS 200mg IV every 4 weeks (q4w) (x3)→100mg SC q8w or GUS 200mg IV q4w (x3)→200mg SC q4w. At the Wk 12 assessment, GUS participants had only received GUS IV.<br>c. Participants assigned to GUS 200mg IV q4w (x3)→100mg SC q8w<br>d.<sup> </sup>Participants assigned to GUS 200mg IV q4w (x3)→200mg SC q4w<br>* Nominal p-value<br></td></tr></tbody></table></div>\n","                <div><strong>Conclusion</strong>:&nbsp;In pooled analyses of the double-blind GALAXI 2 &amp; 3 trials, GUS was efficacious versus PBO in pts with CD regardless of prior biologic therapy exposure in both short-term (Wk12) endpoints and long-term (Wk12 + Wk48) composite endpoints.</div>\n","                                            <div><strong>Nothing to disclose</strong>:&nbsp;No</div>\n","                <div><strong>Disclosure</strong>:&nbsp;<strong >Bruce E. Sands</strong> reports potential conflicts of interest with AbbVie, Abivax, Adiso Therapeutics, Agomab, Alimentiv, Amgen, AnaptysBio, Arena Pharmaceuticals, Artugen Therapeutics, AstraZeneca, Biora Therapeutics, Boehringer Ingelheim, Boston Pharmaceuticals, Bristol Myers Squibb, Calibr, Celgene, Celltrion, ClostraBio, Equillium, Enthera, Evommune, Ferring, Fresenius Kabi, Galapagos, Genentech (Roche), Gilead Sciences, GlaxoSmithKline, Gossamer Bio, Index Pharmaceuticals, Innovation Pharmaceuticals, Inotrem, Janssen, Kaleido, Kallyope, Lilly, Merck, Microbiotica, Mobius Care, Morphic Therapeutics, MRM Health, Pfizer, Nexus Therapeutics, Nimbus Discovery, Odyssey, Therapeutics, Progenity, Prometheus Biosciences, Prometheus Laboratories, Protagonist Therapeutics, Q32 Bio, Rasayana Therapeutics, Recludix Therapeutics, Reistone Biotherapeutics, Sun Pharma, Surrozen, Target RWE, Takeda, Teva, Theravance Biopharma, TLL Pharmaceutical, TR1X, Union Therapeutics and Ventyx Biosciences.<br><strong>Geert D’Haens</strong> reports potential conflicts of interest with AbbVie, Agomab, Alimentiv, AstraZeneca, AMT, Bristol Myers Squibb, Boehringer Ingelheim, Celltrion, Eli Lilly, Exeliom, Galapagos, GlaxoSmithKline, Gossamerbio, Immunic, Index, Johnson &amp; Johnson, Kaleido, Origo, Pfizer, Polpharma, Procise Diagnostics, Prometheus Laboratories, Prometheus Biosciences, Progenity, and Protagonist Therapeutics Inc., Seres Health, and Takeda.<br><strong>Silvio Danese</strong> reports potential conflicts of interest with AbbVie, Alimentiv, Allergan, Amgen, AstraZeneca, Athos, Biogen, Boehringer Ingelheim, Celgene, Celltrion, Eli Lilly, Enthera, Ferring, Gilead, Hospira, Inotrem, Janssen, Johnson &amp; Johnson, MSD, Mundipharma, Mylan, Pfizer, Roche, Sandoz, Sublimity, Takeda, TiGenix, UCB, and Vifor.<br><strong>Tadakazu Hisamatsu</strong> reports potential conflicts of interest with AbbVie, Bristol Myers Squibb, Daiichi-Sankyo, EA Pharma Co, Ltd., Eli Lilly, Gilead Sciences, Janssen Research &amp; Development, LLC., JIMRO, Kissei Pharmaceutical Co, Ltd., Mitsubishi Tanabe Pharma Corporation, Mochida Pharmaceutical Co., Ltd., Nippon Kayaku Co., Ltd., Pfizer Inc., and Takeda Pharmaceutical Co., Ltd.<br><strong>Walter Reinisch</strong> reports potential conflicts of interest with AbbVie, Aesca, Agomab, Algernon, AltruBio, Amgen, AM, AMT, AOP Orphan, Aptalis, Arena, Astellas, AstraZeneca, Avaxia, Roland Berger, Bioclinica, Biogen, Boehringer-Ingelheim, Bristol Myers Squibb, Calyx, Cellerix, Chemocentryx, Celgene, Celltrion, Covance, Danone Austria, DSM, Elan, Eli Lilly, Ernest &amp; Young, Falk, Ferring, Fresenius, Galapagos, Gatehouse, Genentech, Gilead, Grünenthal, ICON, Immundiagnostik, Index, Inova, Intrinsic Imaging, Janssen, Johnson &amp; Johnson, Kyowa Hakko Kirin, Landos, Lipid, LivaNova, Mallinckrodt, Medahead, Medice, MedImmune, Millenium, Mitsubishi Tanabe, MSD, Nash, Nestle, Nippon Kayaku, Novartis, Ocera, OMass, Otsuka, Parexel, PDL, Periconsulting, Pfizer, Pharmacosmos, PLS Education, Procter &amp; Gamble, Prometheus, Protagonist, Provention, Quell, Robarts Clinical Trial, Sandoz, Schering-Plough, Second Genome, Seres, Setpointmedical, Sigmoid, Sublimity, Takeda, Teva, Therakos, Theravance, Tigenix, UCB, Vifor, Yakult, Zealand, Zyngenia, and 4SC.<br><strong>Natalie A. Terry</strong>, <strong>Leonardo Salese</strong>, <strong>Rian Van Rampelbergh</strong>, <strong>Zijiang Yang</strong>, and <strong>Jewel Johanns</strong> are employees of Janssen Research &amp; Development (a Johnson &amp; Johnson Company) and may own company stock/stock options.<br><strong>George DuVall</strong> reports potential conflicts of interest with Janssen.<br><strong>Niazy Abu Farsakh</strong><strong> </strong>reports potential conflicts of interest with Abbott, Amgen, Antanea medical services, Ferring, Hikma, Janssen, and Pfizer.<br><strong>Remo Panaccione </strong>reports potential conflicts of interest with Abbivax, Abbott, AbbVie, Alimentiv (formerly Robarts), Amgen, Arena Pharmaceuticals, AstraZeneca, Biogen, Boehringer Ingelheim, Bristol-Myers Squibb, Celgene, Celltrion, Cosmos Pharmaceuticals, Eisai, Elan, Eli Lilly, Ferring, Fresenius Kabi, Galapagos, Genentech, Gilead Sciences, GlaxoSmithKline, JAMP Bio, Janssen, Merck, Mylan, Novartis, Oppilan Pharma, Organon, Pandion Pharma, Pendopharm, Pfizer, Progenity, Prometheus Biosciences, Protagonist Therapeutics, Roche, Sandoz, Satisfai Health, Shire, Sublimity Therapeutics, Takeda Pharmaceuticals, Theravance Biopharma, Trellus, Viatris, Ventyx, and UCB.</div>\n","        <div>\n","                    <br><strong>Keyword 1</strong>:&nbsp;Crohn's disease&nbsp;&nbsp;&nbsp;\n","                            <strong>Keyword 2</strong>:&nbsp;guselkumab&nbsp;&nbsp;&nbsp;\n","                    </div>\n","  \n","[]\n","<i><u>M. Barreiro-de Acosta</u><sup>1,2</sup>, L. Nieto-Garcia<sup>2</sup>, M. Poncela<sup>3</sup>, C.J. Gargallo-Puyuelo<sup>4</sup>, F. Rodriguez-Moranta<sup>5</sup>, D. Ceballos<sup>6</sup>, M.I. Calvo Moya<sup>7</sup>, S. Garcia-Lopez<sup>8</sup>, M. Aguas<sup>9</sup>, L. Zabalza<sup>10</sup>, C. Martinez Cuevas<sup>11</sup>, R. de Francisco<sup>12</sup>, I. Bastón-Rey<sup>1,13</sup>, F. Bermejo<sup>14</sup>, L. Madero<sup>15</sup>, M.D. Martín-Arranz<sup>16</sup>, M. Rivero<sup>17</sup>, I. Alonso-Abreu<sup>18</sup>, P. Varela Trastoy<sup>19</sup>, M.L. De Castro Parga<sup>20</sup>, B. Camps<sup>21</sup>, C. Martinez Pascual<sup>22</sup>, B. Sicilia<sup>23</sup>, J.P. Gisbert<sup>24</sup>, M. Piqueras<sup>25</sup>, C. Abril<sup>26</sup>, E. Brunet<sup>27</sup>, A. Hernandez Camba<sup>28</sup>, P. Fradejas<sup>29</sup>, Á. Ponferrada-Díaz<sup>30</sup>, M. Mañosa Ciria<sup>31</sup>, M. Algara San Nicolas<sup>32</sup>, J. Castro Poceiro<sup>33</sup>, J.A. Ferrer<sup>34</sup>, E. Ricart<sup>35</sup>, A. Elorza<sup>36</sup>, M.F. García Sepulcre<sup>37</sup>, P. Robledo<sup>38</sup>, M. Teller<sup>39</sup>, E. Sesé<sup>40</sup>, D. Busquets Casals<sup>41</sup>, F. Mesonero Gismero<sup>42</sup>, R. Pajares<sup>43</sup>, L. Bujanda Fernández de Piérola<sup>44</sup>, C. González-Muñoza<sup>45</sup>, D. Martin Rodriguez<sup>46</sup>, M.T. Diz-Lois Palomares<sup>47</sup>, I. Nicolas<sup>48</sup>, E. Domenech<sup>31</sup>, R. Ferreiro-Iglesias<sup>1,2</sup>, ENEIDA project of GETECCU</i><br>\n","<div><sup>1</sup>Complexo Hospitalario Universitario de Santiago, Gastroenterology, Santiago, Spain, <sup>2</sup>Fundación Instituto de Investigación Sanitaria de Santiago de Compostela (FIDIS), Santiago de Compostela, Spain, <sup>3</sup>Hospital General Universitario Gregorio Marañón.Instituto de Investigación Biomédica Gregorio Marañón (IiSGM), Madrid, Spain, <sup>4</sup>Hospital Clínico Universitario “Lozano Blesa”, IIS Aragón and CIBERehd, Zaragoza, Spain, <sup>5</sup>Hospital Universitario de Bellvitge, IDIBELL, Gastroenterology, Barcelona, Spain, <sup>6</sup>Hospital Universitario de Gran Canaria Dr. Negrín, Las Palmas, Spain, <sup>7</sup>Hospital Universitario Puerta de Hierro Majadahonda, Gastroenterology, Madrid, Spain, <sup>8</sup>Hospital Universitario Miguel Servet, Gastroenterology, Zaragoza, Spain, <sup>9</sup>La Fe Universitary And Politecnic Hospital, Gastroenterology Unit, Valencia, Spain, <sup>10</sup>Hospital Universitario de Navarra, Gastroenterology, Pamplona, Spain, <sup>11</sup>Hospital Universitario Rio Hortega, Valladolid, Spain, <sup>12</sup>Hospital Universitario Central de Asturias and ISPA, Gastroenterology, Oviedo, Spain, <sup>13</sup>Fundación Instituto de Investigación Sanitaria de Santiago de Compostela (FIDIS), Santiago de CompostelaSsps, Spain, <sup>14</sup>H Fuenlabrada, Gastroenterology, Fuenlabrada, Spain, <sup>15</sup>Hospital General Universitario de Alicante and CIBERehd, Gastroenterology, Alicante, Spain, <sup>16</sup>Hospital La Paz, Gastroenterology, Madrid, Spain, <sup>17</sup>Marqués de Valdecilla Universitary Hospital, Gastroenterology, Santander, Spain, <sup>18</sup>Hospital Universitario de Canarias (H.U.C.), Santa Cruz de Tenerife, Spain, <sup>19</sup>Hospital de Cabueñes, Gijon, Spain, <sup>20</sup>Hospital Álvaro Cunqueiro, Estrutura Organizativa de Xestión Integrada de Vigo, Gastroenterology, Vigo, Spain, <sup>21</sup>Hospital del Mar, IMIM (Hospital del Mar Medical Research Institute), Gastroenterology, Barcelona, Spain, <sup>22</sup>H.C.U.Virgen de la Arrixaca, Gastroenterology, Murcia, Spain, <sup>23</sup>Hospital Universitario de Burgos, Gastroenterology, Burgos, Spain, <sup>24</sup>Hospital Universitario de La Princesa, Instituto de Investigación Sanitaria Princesa (IIS-Princesa), Universidad Autónoma de Madrid (UAM), and Centro de Investigación Biomédica en Red de Enfermedades Hepáticas y Digestivas (CIBERehd), Gastroenterology, Madrid, Spain, <sup>25</sup>Consorci Sanitari Terrasa, Gastroenterology, Barcelona, Spain, <sup>26</sup>Hospital Clínico Universitario de Valencia, Gastroenterology, Valencia, Spain, <sup>27</sup>Parc Tauli, Digestive, Sabadell, Spain, <sup>28</sup>Complejo Hospitalario Universitario Ntra. Sra. de Candelaria, Gastroenterology, Santa Cruz de Tenerife, Spain, <sup>29</sup>Complejo Asistencial de Zamora, Gastroenterology, Zamora, Spain, <sup>30</sup>Hospital Infanta Leonor, Madrid, Spain, <sup>31</sup>Hospital Universitari Germans Trias I Pujol and CIBEREHD, Gastroenterology, Badalona, Spain, <sup>32</sup>H.12 Octubre, Madrid, Spain, <sup>33</sup>Hospital de Sant Joan Despí Moisès Broggi, Gastroenterology, Barcelona, Spain, <sup>34</sup>Hospital Fundacion Alcorcon, Gastroenterology, Alcorcon, Spain, <sup>35</sup>Hospital Clínic i Provincial, Institut d'Investigacions Biomèdiques August Pi i Sunyer, Gastroenterology, Barcelona, Spain, <sup>36</sup>Hospital de Galdakao-Usansolo, Gastroenterology, Galdakao, Spain, <sup>37</sup>Hospital General Universitario de Elche, Gastroenterology, Elche, Spain, <sup>38</sup>Hospital Universitario de Cáceres, Caceres, Spain, <sup>39</sup>Althaia Xarxa Assistencial universitaria de Manresa, Gastroenterology, Manresa, Spain, <sup>40</sup>Hospital Universitario Arnau de Vilanova, Gastroenterology, Lleida, Spain, <sup>41</sup>H. DE GIRONA DR. J. TRUETA/ICO GERONA, Gastroenterology, Girona, Spain, <sup>42</sup>Hospital Ramon y Cajal, Gastroenterology, Madrid, Spain, <sup>43</sup>Hospital Universitario Infanta Sofía, Gastroenterology, Madrid, Spain, <sup>44</sup>Hospital Universitario Donostia and  Biodonostia, San Sebastián, Spain, <sup>45</sup>Hospital Santa Creu i Sant Pau, Gastroenterology, Barcelona, Spain, <sup>46</sup>Hospital Infanta Cristina, Madrid, Spain, <sup>47</sup>Hospital Universitario A Coruña, Gastroenterology, A Coruña, Spain, <sup>48</sup>Hospital General Universitario Reina Sofía, Gastroenterology, Murcia, Spain<br><br></div>\n","\n","            <div><strong>Introduction</strong>:&nbsp;Phase III trials have demonstrated the efficacy of risankizumab (RZB), this being the first humanised monoclonal IgG1 antibody which targets the interleukin 23 p19 subunit, in moderate-to-severe Crohn's disease (CD). However, real-world data in induction with this drug is scarce. The aim of the study was to assess the real-world short-term effectiveness and safety of RZB in multi-refractory patients with CD.</div>\n","                <div><strong>Aims & Methods</strong>:&nbsp;Adult CD patients that had received induction doses with RZB in the ENEIDA registry—a large prospectively maintained Spanish database promoted by the Spanish Working Group in Crohn’s and Colitis (GETECCU)— were included. Clinical and demographical data including number of previous biologics and advanced therapies were included. The primary endpoint was steroid-free clinical remission evaluated at any point between weeks 8 and 12 (Harvey-Bradshaw [HB] score &lt;5). Secondary endpoints included steroid free clinical remission at the end of follow-up, clinical, biochemical improvement (C-reactive protein (CRP) and/or faecal calprotectine (FC)), need for CD-related surgery and adverse events related with the drug. The data are presented as median (range), and frequencies and percentages. Statistical analysis was performed using the Chi-square or Fisher's test, and the Student-<em >t</em> test or U-Mann-Whitney test as appropriate.</div>\n","                <div><strong>Results</strong>:&nbsp;417 CD patients (52% male) were included: median age was 50 (18-84) years, median disease duration 14 years, 27% were smokers. Regarding Montreal classification, 49% were L1, 10% L2, and 38% L3. Most of the patients presented inflammatory pattern (47%), 32% presented structuring, 21% penetrating behavior and perianal disease was present in 24%. 48% had abdominal CD surgery and 41% presented extraintestinal manifestations. Only 8% received concomitant immunossupressives with RZB. Regarding previous exposure to biologics or JAK inhibitors, only 3% were naïve, 49% had been exposed to 1-2 advanced therapies and 48% to 3 or more biologics or JAKs. After induction 58% of the patients achieved steroid-free clinical remission and at the end of the follow-up steroid-free clinical remission was observed in 62% of the patients. Median FC decreased from week 530 μg/g (week 0) to 333 μg/g after induction (p &lt;0.001). Regarding CRP, this decreased from median 7 mg/L at week 0 to 4.7 mg/L after induction (p = 0.002). 15 patients (3.5%) underwent intestinal resection. 23 patients (5.5%) presented adverse events including 6 infections.</div>\n","                <div><strong>Conclusion</strong>:&nbsp;This is the largest cohort study of real-world data with RZB, and it includes highly refractory patients with CD and multiple prior drug failures. RZB induction provided a steroid-free clinical remission in nearly 60% of patients and a significative improvement in biomarkers. Safety was consistent with the known profile of RZB in previous trials.</div>\n","                                            <div><strong>Nothing to disclose</strong>:&nbsp;No</div>\n","                <div><strong>Disclosure</strong>:&nbsp;MBA has served as a speaker, consultant and advisory member for or has received research funding from MSD, AbbVie, Janssen, Kern Pharma, Celltrion, Takeda, GALAPAGOS, Pfizer, Sandoz, Biogen, Fresenius, Lilly, Ferring, Faes Farma, Dr. Falk Pharma, Chiesi, Adacyte and TillotsPharma.</div>\n","        <div>\n","                    <br><strong>Keyword 1</strong>:&nbsp;Risankizumab&nbsp;&nbsp;&nbsp;\n","                            <strong>Keyword 2</strong>:&nbsp;Crohn&nbsp;&nbsp;&nbsp;\n","                            <strong>Keyword 3</strong>:&nbsp;Induction\n","            </div>\n","  \n","[{'Session Date': '2024-10-14', 'Session Start Time': '08:30', 'Session End Time': '09:30', 'Session Timezone': 'CEST', 'Session Location': 'A3', 'Session Number': 80, 'Session Pathways': 'IBD', 'Session CME': '', 'Session Type': 'Abstract Session', 'Session Group': 3, 'Session Title': 'Ameliorating acute disease through novel treatment in IBD: Part 1', 'Session Description': None, 'Session Format': 'Live, Recording, Q&A', 'Presentation Date': '2024-10-14', 'Presentation Location': 'A3', 'Presentation Start Time': '08:30', 'Presentation End Time': '08:42', 'Presentation ID': 702, 'Presentation Number': 'OP042', 'Presentation Title': 'EFFICACY OF GUSELKUMAB VERSUS PLACEBO IN CROHN’S DISEASE BASED ON PRIOR RESPONSE/EXPOSURE TO BIOLOGIC THERAPY: RESULTS OF THE GALAXI 2 &AMP; 3 PHASE 3 STUDIES', 'Presentation Title (lowercase)': 'efficacy of guselkumab versus placebo in crohn’s disease based on prior response/exposure to biologic therapy: results of the galaxi 2 &amp; 3 phase 3 studies', 'Abstract Authors': 'Bruce E. Sands', 'Abstract Affiliations': 'Icahn School of Medicine at Mount Sinai', 'Abstract': '<i><u>B.E. Sands</u><sup>1</sup>, G. D’Haens<sup>2</sup>, S. Danese<sup>3</sup>, T. Hisamatsu<sup>4</sup>, W. Reinisch<sup>5</sup>, N.A. Terry<sup>6</sup>, L. Salese<sup>6</sup>, R. Van Rampelbergh<sup>7</sup>, Z. Yang<sup>6</sup>, J. Johanns<sup>6</sup>, G. DuVall<sup>8</sup>, N. Abu Farsakh<sup>9</sup>, R. Panaccione<sup>10</sup></i><br>\\r\\n<div><sup>1</sup>Dr. Henry D Janowitz Division of Gastroenterology, Icahn School of Medicine at Mount Sinai, New York, United States of America, <sup>2</sup>Department of Gastroenterology, Amsterdam University Medical Centers, Amsterdam, Netherlands, <sup>3</sup>Gastroenterology and Endoscopy, IRCCS San Raffaele Hospital and Vita-Salute San Raffaele University, Milan, Italy, <sup>4</sup>Department of Gastroenterology and Hepatology, Kyorin University, Tokyo, Japan, <sup>5</sup>Division of Gastroenterology & Hepatology, Medical University of Vienna, Vienna, Austria, <sup>6</sup>Janssen Research & Development, LLC, Spring House, United States of America, <sup>7</sup>Janssen Research & Development, Antwerp, Belgium, <sup>8</sup>Tyler Research Institute, LLC, Tyler, United States of America, <sup>9</sup>King Abdullah University Hospital, Irbid, Jordan, <sup>10</sup>Inflammatory Bowel Disease Unit, Division of Gastroenterology and Hepatology, University of Calgary, Calgary, Canada<br><br></div>\\r\\n\\r\\n            <div><strong>Introduction</strong>:&nbsp;GALAXI 2 &amp; 3 are identically designed 48-week, randomized, double-blind, double-dummy, placebo (PBO)- and active-comparator, treat-through registrational trials assessing the efficacy and safety of IV induction and SC maintenance therapy with guselkumab (GUS), a dual-acting IL-23p19 subunit inhibitor, in participants (pts) with moderately to severely active Crohn’s disease (CD). Primary results of each study were previously reported; here we explore the impact of biologic therapy exposure on efficacy outcomes with GUS compared to PBO in the pooled GALAXI 2 &amp; 3 dataset.</div>\\r\\n                <div><strong>Aims & Methods</strong>:&nbsp;Pts with inadequate response or intolerance to conventional therapy or biologic therapy (BIO-ir), active CD (CDAI 220–450 + mean daily SF&gt;3 or AP&gt;1), and SES-CD≥6 (≥4 for isolated ileal disease) were randomly assigned 2:2:2:1 at week (Wk) 0 to GUS 200mg IV q4w (x3)→100mg SC q8w, GUS 200mg IV q4w (x3)→200mg SC q4w, UST IV→SC, or PBO. In each trial, the composite co-primary endpoints were 1) clinical response at Wk12 + clinical remission at Wk48, and 2) clinical response at Wk12 + endoscopic response at Wk48, comparing each GUS regimen to PBO. Major secondary endpoints included clinical remission at Wk12 and endoscopic response at Wk12. Analyses of BIO-ir and BIO-naïve subgroups in the individual trials were prespecified; pooled analyses were performed post hoc.</div>\\r\\n                <div><strong>Results</strong>:&nbsp;\\u200bIn the pooled dataset, 52% (378/730) of pts randomized to PBO or GUS had a prior history of BIO-ir and 42% (305/730) were BIO-naïve (6% [47/730] had prior exposure to biologics but no documented failure). At Wk12, GUS achieved higher rates of clinical remission and endoscopic response compared to PBO in the overall population and BIO-ir and BIO-naïve subgroups (<strong>Table</strong>). For the composite co-primary endpoints, both GUS regimens also achieved higher rates of 1) clinical response at Wk12 + clinical remission at Wk48 and 2) clinical response at Wk12 + endoscopic response at Wk48 compared to PBO in the overall population and BIO-ir and BIO-naïve subgroups.<br><br><table style=\"width: 100%; margin-left: calc(0%);\" class=\"\"><tbody><tr><td style=\"width: 100%;\" colspan=\"10\" class=\"fr-selected-cell\"><strong>Ta</strong><strong>ble. Efficacy of guselkumab (GUS) and placebo (PBO) for short- and long-term clinical and endoscopic endpoints in all participants and BIO-ir and BIO-naïve subgroups in the pooled GALAXI 2 &amp; 3 dataset</strong><br></td></tr><tr><td style=\"width: 10.0069%;\" class=\"\"><br></td><td style=\"width: 30%; text-align: center;\" colspan=\"3\"><strong>Overall population</strong><br></td><td style=\"width: 29.9523%; text-align: center;\" colspan=\"3\"><strong>Participants with a history of inadequate response or intolerance to at least 1 prior biologic<sup>a</sup> (BIO-ir)</strong><br></td><td style=\"width: 30%; text-align: center;\" colspan=\"3\"><strong>Participants with no history of exposure to biologics<sup>a</sup> (BIO-naïve) </strong><br></td></tr><tr><td style=\"width: 10.0068%; background-color: rgb(239, 239, 239);\" class=\"\"><br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=148<br></td><td style=\"width: 20%; background-color: rgb(239, 239, 239); text-align: center;\" colspan=\"2\"><strong>GUS 200mg IV combined</strong><sup><strong>b</strong></sup><br>N=582<br></td><td style=\"width: 10.0068%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=78<br></td><td style=\"width: 20%; background-color: rgb(239, 239, 239); text-align: center;\" colspan=\"2\"><strong>GUS 200mg IV combined</strong><sup><strong>b</strong></sup><br>N=300<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=61<br></td><td style=\"width: 20%; background-color: rgb(239, 239, 239); text-align: center;\" colspan=\"2\"><strong>GUS 200mg IV combined</strong><sup><strong>b</strong></sup><br>N=244<br></td></tr><tr><td style=\"width: 10.0068%;\"><strong>Clinical remission at Wk12</strong><br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\">28 (18.9%)<br></td><td style=\"width: 20%; text-align: center;\" colspan=\"2\">274 (47.1%)<br><strong>∆ 28.0%</strong><br>P&lt;.001*<br></td><td style=\"width: 10.0068%; text-align: center; vertical-align: middle;\" class=\"\">15 (19.2%)<br></td><td style=\"width: 20%; text-align: center;\" colspan=\"2\">138 (46.0%)<br><strong>∆ 26.9%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">10 (16.4%)<br></td><td style=\"width: 20%; text-align: center;\" colspan=\"2\">121 (49.6%)<br><strong>∆ 32.1%</strong><br>P&lt;.001*<br></td></tr><tr><td style=\"width: 10.0068%;\"><strong>Endoscopic response at Wk12</strong><br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">18 (12.2%)<br></td><td style=\"width: 20.0136%; text-align: center;\" colspan=\"2\">215 (36.9%)<br><strong>∆ 25.2%</strong><br>P&lt;.001*<br></td><td style=\"width: 10.0068%; text-align: center; vertical-align: middle;\" class=\"\">5 (6.4%)<br></td><td style=\"width: 20.0136%; text-align: center;\" colspan=\"2\">87 (29.0%)<br><strong>∆ 22.6%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">11 (18.0%)<br></td><td style=\"width: 20.0136%; text-align: center;\" colspan=\"2\">113 (46.3%)<br><strong>∆ 27.7%</strong><br>P&lt;.001*<br></td></tr><tr><td style=\"width: 10.0068%; background-color: rgb(239, 239, 239);\" class=\"\"><br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=148<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 100mg</strong><sup><strong>c</strong></sup><br>N=286<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 200mg</strong><sup><strong>d</strong></sup><br>N=296<br></td><td style=\"width: 10.0068%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=78<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 100mg</strong><sup><strong>c</strong></sup><br>N=153<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 200mg</strong><sup><strong>d</strong></sup><br>N=147<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>PBO</strong><br>N=61<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 100mg</strong><sup><strong>c</strong></sup><br>N=116<br></td><td style=\"width: 10%; background-color: rgb(239, 239, 239); text-align: center;\"><strong>GUS 200mg</strong><sup><strong>d</strong></sup><br>N=128<br></td></tr><tr><td style=\"width: 10.0068%;\"><strong>Clinical response at Wk12 + clinical remission at Wk48<br></strong>(composite co-primary endpoint)<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">18 (12.2%)<br></td><td style=\"width: 10%; text-align: center;\">137 (47.9%)<br><strong>∆ 35.8%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">152 (51.4%)<br><strong>∆ 39.0%</strong><br>P&lt;.001*<br></td><td style=\"width: 10.0068%; text-align: center; vertical-align: middle;\" class=\"\">10 (12.8%)<br></td><td style=\"width: 10%; text-align: center;\">70 (45.8%)<br><strong>∆ 33.2%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">73 (49.7%)<br><strong>∆ 37.4%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">7 (11.5%)<br></td><td style=\"width: 10%; text-align: center;\">60 (51.7%)<br><strong>∆ 40.4%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">70 (54.7%)<br><strong>∆ 41.8%</strong><br>P&lt;.001*<br></td></tr><tr><td style=\"width: 10.0068%;\"><strong>Clinical response at Wk12 + endoscopic response at Wk48</strong><br>(composite co-primary endpoint)<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">8 (5.4%)<br></td><td style=\"width: 10%; text-align: center;\">104 (36.4%)<br><strong>∆ 30.7%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">110 (37.2%)<br><strong>∆ 31.8%</strong><br>P&lt;.001*<br></td><td style=\"width: 10.0068%; text-align: center; vertical-align: middle;\" class=\"\">4 (5.1%)<br></td><td style=\"width: 10%; text-align: center;\">55 (35.9%)<br><strong>∆ 30.9%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">46 (31.3%)<br><strong>∆ 26.7%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center; vertical-align: middle;\" class=\"\">4 (6.6%)<br></td><td style=\"width: 10%; text-align: center;\">47 (40.5%)<br><strong>∆ 33.0%</strong><br>P&lt;.001*<br></td><td style=\"width: 10%; text-align: center;\">56 (43.8%)<br><strong>∆ 36.5%</strong><br>P&lt;.001*<br></td></tr><tr><td style=\"width: 100%;\" colspan=\"10\">Data are presented as n (%); ∆% (adjusted treatment difference) vs PBO; p-value vs PBO.<br>Adjusted treatment differences and p-values were based on the common risk difference by use of Mantel-Haenszel stratum weights and the Sato variance estimator. The stratification variables used are baseline CDAI score (≤300 or &gt;300), baseline SES-CD score (≤12 or &gt;12), BIO-ir status (Yes or No; this variable used only in analyses of the overall population), and baseline corticosteroid use (Yes or No). Population N values reflect the primary analysis set, defined as all randomized participants who received at least 1 (partial or complete) dose of study intervention and had a screening SES-CD score ≥6 (or ≥4 for participants with isolated ileal disease). PBO pts not in clinical response at week (Wk) 12 switched to UST.<br><br><u>Endpoint definitions</u>: clinical remission, CDAI &lt; 150; endoscopic response, ≥50% improvement from baseline in SES-CD or SES-CD ≤ 2; clinical response, ≥100-point reduction from baseline in CDAI or CDAI &lt; 150. Participants with treatment failure or missing data were considered to not have met the endpoint.<br><br>a. Anti-TNF agents or vedolizumab<br>b. Participants assigned to either GUS 200mg IV every 4 weeks (q4w) (x3)→100mg SC q8w or GUS 200mg IV q4w (x3)→200mg SC q4w. At the Wk 12 assessment, GUS participants had only received GUS IV.<br>c. Participants assigned to GUS 200mg IV q4w (x3)→100mg SC q8w<br>d.<sup> </sup>Participants assigned to GUS 200mg IV q4w (x3)→200mg SC q4w<br>* Nominal p-value<br></td></tr></tbody></table></div>\\r\\n                <div><strong>Conclusion</strong>:&nbsp;In pooled analyses of the double-blind GALAXI 2 &amp; 3 trials, GUS was efficacious versus PBO in pts with CD regardless of prior biologic therapy exposure in both short-term (Wk12) endpoints and long-term (Wk12 + Wk48) composite endpoints.</div>\\r\\n                                            <div><strong>Nothing to disclose</strong>:&nbsp;No</div>\\r\\n                <div><strong>Disclosure</strong>:&nbsp;<strong >Bruce E. Sands</strong> reports potential conflicts of interest with AbbVie, Abivax, Adiso Therapeutics, Agomab, Alimentiv, Amgen, AnaptysBio, Arena Pharmaceuticals, Artugen Therapeutics, AstraZeneca, Biora Therapeutics, Boehringer Ingelheim, Boston Pharmaceuticals, Bristol Myers Squibb, Calibr, Celgene, Celltrion, ClostraBio, Equillium, Enthera, Evommune, Ferring, Fresenius Kabi, Galapagos, Genentech (Roche), Gilead Sciences, GlaxoSmithKline, Gossamer Bio, Index Pharmaceuticals, Innovation Pharmaceuticals, Inotrem, Janssen, Kaleido, Kallyope, Lilly, Merck, Microbiotica, Mobius Care, Morphic Therapeutics, MRM Health, Pfizer, Nexus Therapeutics, Nimbus Discovery, Odyssey, Therapeutics, Progenity, Prometheus Biosciences, Prometheus Laboratories, Protagonist Therapeutics, Q32 Bio, Rasayana Therapeutics, Recludix Therapeutics, Reistone Biotherapeutics, Sun Pharma, Surrozen, Target RWE, Takeda, Teva, Theravance Biopharma, TLL Pharmaceutical, TR1X, Union Therapeutics and Ventyx Biosciences.<br><strong>Geert D’Haens</strong> reports potential conflicts of interest with AbbVie, Agomab, Alimentiv, AstraZeneca, AMT, Bristol Myers Squibb, Boehringer Ingelheim, Celltrion, Eli Lilly, Exeliom, Galapagos, GlaxoSmithKline, Gossamerbio, Immunic, Index, Johnson &amp; Johnson, Kaleido, Origo, Pfizer, Polpharma, Procise Diagnostics, Prometheus Laboratories, Prometheus Biosciences, Progenity, and Protagonist Therapeutics Inc., Seres Health, and Takeda.<br><strong>Silvio Danese</strong> reports potential conflicts of interest with AbbVie, Alimentiv, Allergan, Amgen, AstraZeneca, Athos, Biogen, Boehringer Ingelheim, Celgene, Celltrion, Eli Lilly, Enthera, Ferring, Gilead, Hospira, Inotrem, Janssen, Johnson &amp; Johnson, MSD, Mundipharma, Mylan, Pfizer, Roche, Sandoz, Sublimity, Takeda, TiGenix, UCB, and Vifor.<br><strong>Tadakazu Hisamatsu</strong> reports potential conflicts of interest with AbbVie, Bristol Myers Squibb, Daiichi-Sankyo, EA Pharma Co, Ltd., Eli Lilly, Gilead Sciences, Janssen Research &amp; Development, LLC., JIMRO, Kissei Pharmaceutical Co, Ltd., Mitsubishi Tanabe Pharma Corporation, Mochida Pharmaceutical Co., Ltd., Nippon Kayaku Co., Ltd., Pfizer Inc., and Takeda Pharmaceutical Co., Ltd.<br><strong>Walter Reinisch</strong> reports potential conflicts of interest with AbbVie, Aesca, Agomab, Algernon, AltruBio, Amgen, AM, AMT, AOP Orphan, Aptalis, Arena, Astellas, AstraZeneca, Avaxia, Roland Berger, Bioclinica, Biogen, Boehringer-Ingelheim, Bristol Myers Squibb, Calyx, Cellerix, Chemocentryx, Celgene, Celltrion, Covance, Danone Austria, DSM, Elan, Eli Lilly, Ernest &amp; Young, Falk, Ferring, Fresenius, Galapagos, Gatehouse, Genentech, Gilead, Grünenthal, ICON, Immundiagnostik, Index, Inova, Intrinsic Imaging, Janssen, Johnson &amp; Johnson, Kyowa Hakko Kirin, Landos, Lipid, LivaNova, Mallinckrodt, Medahead, Medice, MedImmune, Millenium, Mitsubishi Tanabe, MSD, Nash, Nestle, Nippon Kayaku, Novartis, Ocera, OMass, Otsuka, Parexel, PDL, Periconsulting, Pfizer, Pharmacosmos, PLS Education, Procter &amp; Gamble, Prometheus, Protagonist, Provention, Quell, Robarts Clinical Trial, Sandoz, Schering-Plough, Second Genome, Seres, Setpointmedical, Sigmoid, Sublimity, Takeda, Teva, Therakos, Theravance, Tigenix, UCB, Vifor, Yakult, Zealand, Zyngenia, and 4SC.<br><strong>Natalie A. Terry</strong>, <strong>Leonardo Salese</strong>, <strong>Rian Van Rampelbergh</strong>, <strong>Zijiang Yang</strong>, and <strong>Jewel Johanns</strong> are employees of Janssen Research &amp; Development (a Johnson &amp; Johnson Company) and may own company stock/stock options.<br><strong>George DuVall</strong> reports potential conflicts of interest with Janssen.<br><strong>Niazy Abu Farsakh</strong><strong> </strong>reports potential conflicts of interest with Abbott, Amgen, Antanea medical services, Ferring, Hikma, Janssen, and Pfizer.<br><strong>Remo Panaccione </strong>reports potential conflicts of interest with Abbivax, Abbott, AbbVie, Alimentiv (formerly Robarts), Amgen, Arena Pharmaceuticals, AstraZeneca, Biogen, Boehringer Ingelheim, Bristol-Myers Squibb, Celgene, Celltrion, Cosmos Pharmaceuticals, Eisai, Elan, Eli Lilly, Ferring, Fresenius Kabi, Galapagos, Genentech, Gilead Sciences, GlaxoSmithKline, JAMP Bio, Janssen, Merck, Mylan, Novartis, Oppilan Pharma, Organon, Pandion Pharma, Pendopharm, Pfizer, Progenity, Prometheus Biosciences, Protagonist Therapeutics, Roche, Sandoz, Satisfai Health, Shire, Sublimity Therapeutics, Takeda Pharmaceuticals, Theravance Biopharma, Trellus, Viatris, Ventyx, and UCB.</div>\\r\\n        <div>\\r\\n                    <br><strong>Keyword 1</strong>:&nbsp;Crohn\\'s disease&nbsp;&nbsp;&nbsp;\\r\\n                            <strong>Keyword 2</strong>:&nbsp;guselkumab&nbsp;&nbsp;&nbsp;\\r\\n                    </div>\\r\\n  ', 'Abstract Url': 'https://programme.ueg.eu/week2024/#/details/presentations/1077', 'Company': None, 'Asset': 'Guselkumab', 'Indication': 'Crohn', 'MoA': None, 'Other Keywords': None}]\n"]},{"ename":"KeyboardInterrupt","evalue":"","output_type":"error","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)","Cell \u001b[0;32mIn[14], line 42\u001b[0m\n\u001b[1;32m     40\u001b[0m Abstract_url \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhttps://programme.ueg.eu/week2024/#/details/presentations/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mAbstract_id\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     41\u001b[0m Abstract_endpoint \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhttps://ueg2024.abstract.documedias.systems/api/v1/manager/abstract/multi/html/id/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mAbstract_id\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/template/planner_preview?program_type=week&system=program_week_2024\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m---> 42\u001b[0m responseAbstract \u001b[38;5;241m=\u001b[39m \u001b[43mrequests\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43mAbstract_endpoint\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     43\u001b[0m dataAbstract \u001b[38;5;241m=\u001b[39m responseAbstract\u001b[38;5;241m.\u001b[39mjson()\n\u001b[1;32m     45\u001b[0m abstracts \u001b[38;5;241m=\u001b[39m dataAbstract\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mAbstract_id\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/requests/api.py:73\u001b[0m, in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     62\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(url, params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m     63\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[1;32m     64\u001b[0m \n\u001b[1;32m     65\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 73\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mget\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mparams\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/requests/api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[1;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[1;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[1;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[0;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43msession\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/requests/sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    584\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    585\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[1;32m    586\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[1;32m    587\u001b[0m }\n\u001b[1;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprep\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43msend_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/requests/sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    700\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[1;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m \u001b[43madapter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[1;32m    706\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/requests/adapters.py:667\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    664\u001b[0m     timeout \u001b[38;5;241m=\u001b[39m TimeoutSauce(connect\u001b[38;5;241m=\u001b[39mtimeout, read\u001b[38;5;241m=\u001b[39mtimeout)\n\u001b[1;32m    666\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 667\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    668\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    669\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    670\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    671\u001b[0m \u001b[43m        \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    672\u001b[0m \u001b[43m        \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    673\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    674\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    675\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    676\u001b[0m \u001b[43m        \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    677\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    678\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    679\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    681\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m    682\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(err, request\u001b[38;5;241m=\u001b[39mrequest)\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py:789\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[1;32m    786\u001b[0m response_conn \u001b[38;5;241m=\u001b[39m conn \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m release_conn \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m    788\u001b[0m \u001b[38;5;66;03m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[0;32m--> 789\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    790\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    791\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    792\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    793\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    794\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    795\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    796\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    797\u001b[0m \u001b[43m    \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mretries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    798\u001b[0m \u001b[43m    \u001b[49m\u001b[43mresponse_conn\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse_conn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    799\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpreload_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    800\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdecode_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    801\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mresponse_kw\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    802\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    804\u001b[0m \u001b[38;5;66;03m# Everything went great!\u001b[39;00m\n\u001b[1;32m    805\u001b[0m clean_exit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py:466\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[1;32m    463\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    464\u001b[0m     \u001b[38;5;66;03m# Trigger any extra validation we need to do.\u001b[39;00m\n\u001b[1;32m    465\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 466\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate_conn\u001b[49m\u001b[43m(\u001b[49m\u001b[43mconn\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    467\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[1;32m    468\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_raise_timeout(err\u001b[38;5;241m=\u001b[39me, url\u001b[38;5;241m=\u001b[39murl, timeout_value\u001b[38;5;241m=\u001b[39mconn\u001b[38;5;241m.\u001b[39mtimeout)\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py:1095\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[0;34m(self, conn)\u001b[0m\n\u001b[1;32m   1093\u001b[0m \u001b[38;5;66;03m# Force connect early to allow us to validate the connection.\u001b[39;00m\n\u001b[1;32m   1094\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_closed:\n\u001b[0;32m-> 1095\u001b[0m     \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1097\u001b[0m \u001b[38;5;66;03m# TODO revise this, see https://github.com/urllib3/urllib3/issues/2791\u001b[39;00m\n\u001b[1;32m   1098\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_verified \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mproxy_is_verified:\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/urllib3/connection.py:730\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    727\u001b[0m     \u001b[38;5;66;03m# Remove trailing '.' from fqdn hostnames to allow certificate validation\u001b[39;00m\n\u001b[1;32m    728\u001b[0m     server_hostname_rm_dot \u001b[38;5;241m=\u001b[39m server_hostname\u001b[38;5;241m.\u001b[39mrstrip(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m.\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m--> 730\u001b[0m     sock_and_verified \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_and_match_hostname\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    731\u001b[0m \u001b[43m        \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    732\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcert_reqs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_reqs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    733\u001b[0m \u001b[43m        \u001b[49m\u001b[43mssl_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    734\u001b[0m \u001b[43m        \u001b[49m\u001b[43mssl_minimum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_minimum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    735\u001b[0m \u001b[43m        \u001b[49m\u001b[43mssl_maximum_version\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_maximum_version\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    736\u001b[0m \u001b[43m        \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    737\u001b[0m \u001b[43m        \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    738\u001b[0m \u001b[43m        \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    739\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcert_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    740\u001b[0m \u001b[43m        \u001b[49m\u001b[43mkey_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    741\u001b[0m \u001b[43m        \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    742\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname_rm_dot\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    743\u001b[0m \u001b[43m        \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mssl_context\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    744\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    745\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    746\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_fingerprint\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43massert_fingerprint\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    747\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    748\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock \u001b[38;5;241m=\u001b[39m sock_and_verified\u001b[38;5;241m.\u001b[39msocket\n\u001b[1;32m    750\u001b[0m \u001b[38;5;66;03m# If an error occurs during connection/handshake we may need to release\u001b[39;00m\n\u001b[1;32m    751\u001b[0m \u001b[38;5;66;03m# our lock so another connection can probe the origin.\u001b[39;00m\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/urllib3/connection.py:909\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_and_match_hostname\u001b[0;34m(sock, cert_reqs, ssl_version, ssl_minimum_version, ssl_maximum_version, cert_file, key_file, key_password, ca_certs, ca_cert_dir, ca_cert_data, assert_hostname, assert_fingerprint, server_hostname, ssl_context, tls_in_tls)\u001b[0m\n\u001b[1;32m    906\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m is_ipaddress(normalized):\n\u001b[1;32m    907\u001b[0m         server_hostname \u001b[38;5;241m=\u001b[39m normalized\n\u001b[0;32m--> 909\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43mssl_wrap_socket\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    910\u001b[0m \u001b[43m    \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    911\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkeyfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    912\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcertfile\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcert_file\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    913\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_password\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkey_password\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    914\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_certs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_certs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    915\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_dir\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_dir\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    916\u001b[0m \u001b[43m    \u001b[49m\u001b[43mca_cert_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mca_cert_data\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    917\u001b[0m \u001b[43m    \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    918\u001b[0m \u001b[43m    \u001b[49m\u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    919\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    920\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    922\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    923\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m assert_fingerprint:\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/urllib3/util/ssl_.py:469\u001b[0m, in \u001b[0;36mssl_wrap_socket\u001b[0;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data, tls_in_tls)\u001b[0m\n\u001b[1;32m    465\u001b[0m         context\u001b[38;5;241m.\u001b[39mload_cert_chain(certfile, keyfile, key_password)\n\u001b[1;32m    467\u001b[0m context\u001b[38;5;241m.\u001b[39mset_alpn_protocols(ALPN_PROTOCOLS)\n\u001b[0;32m--> 469\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m \u001b[43m_ssl_wrap_socket_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcontext\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtls_in_tls\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    470\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ssl_sock\n","File \u001b[0;32m~/Desktop/Webscraping/.venv/lib/python3.12/site-packages/urllib3/util/ssl_.py:513\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_impl\u001b[0;34m(sock, ssl_context, tls_in_tls, server_hostname)\u001b[0m\n\u001b[1;32m    510\u001b[0m     SSLTransport\u001b[38;5;241m.\u001b[39m_validate_ssl_context_for_tls_in_tls(ssl_context)\n\u001b[1;32m    511\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m SSLTransport(sock, ssl_context, server_hostname)\n\u001b[0;32m--> 513\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mssl_context\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwrap_socket\u001b[49m\u001b[43m(\u001b[49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m)\u001b[49m\n","File \u001b[0;32m/usr/lib/python3.12/ssl.py:455\u001b[0m, in \u001b[0;36mSSLContext.wrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[1;32m    449\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrap_socket\u001b[39m(\u001b[38;5;28mself\u001b[39m, sock, server_side\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[1;32m    450\u001b[0m                 do_handshake_on_connect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    451\u001b[0m                 suppress_ragged_eofs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    452\u001b[0m                 server_hostname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, session\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    453\u001b[0m     \u001b[38;5;66;03m# SSLSocket class handles server_hostname encoding before it calls\u001b[39;00m\n\u001b[1;32m    454\u001b[0m     \u001b[38;5;66;03m# ctx._wrap_socket()\u001b[39;00m\n\u001b[0;32m--> 455\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msslsocket_class\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_create\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    456\u001b[0m \u001b[43m        \u001b[49m\u001b[43msock\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msock\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    457\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_side\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_side\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    458\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdo_handshake_on_connect\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    459\u001b[0m \u001b[43m        \u001b[49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msuppress_ragged_eofs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    460\u001b[0m \u001b[43m        \u001b[49m\u001b[43mserver_hostname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mserver_hostname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    461\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontext\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    462\u001b[0m \u001b[43m        \u001b[49m\u001b[43msession\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msession\u001b[49m\n\u001b[1;32m    463\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n","File \u001b[0;32m/usr/lib/python3.12/ssl.py:1042\u001b[0m, in \u001b[0;36mSSLSocket._create\u001b[0;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[1;32m   1039\u001b[0m             \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0.0\u001b[39m:\n\u001b[1;32m   1040\u001b[0m                 \u001b[38;5;66;03m# non-blocking\u001b[39;00m\n\u001b[1;32m   1041\u001b[0m                 \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdo_handshake_on_connect should not be specified for non-blocking sockets\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m-> 1042\u001b[0m             \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdo_handshake\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1043\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m:\n\u001b[1;32m   1044\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n","File \u001b[0;32m/usr/lib/python3.12/ssl.py:1320\u001b[0m, in \u001b[0;36mSSLSocket.do_handshake\u001b[0;34m(self, block)\u001b[0m\n\u001b[1;32m   1318\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0.0\u001b[39m \u001b[38;5;129;01mand\u001b[39;00m block:\n\u001b[1;32m   1319\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(\u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[0;32m-> 1320\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sslobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdo_handshake\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1321\u001b[0m \u001b[38;5;28;01mfinally\u001b[39;00m:\n\u001b[1;32m   1322\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(timeout)\n","\u001b[0;31mKeyboardInterrupt\u001b[0m: "]}],"source":["\n","# Define the fields you want to extract for sessions and presentations combined\n","fields = [\n","    \"Session Date\", \"Session Start Time\", \"Session End Time\", \"Session Timezone\",\n","    \"Session Location\", \"Session Number\", \"Session Pathways\", \"Session Format\",\n","    \"Session CME\", \"Session Type\", \"Session Group\", \"Session Title\", \"Session Description\",\n","    \"Presentation Date\", \"Presentation Location\", \"Presentation Start Time\", \"Presentation End Time\",\n","    \"Presentation ID\", \"Presentation Number\", \"Presentation Title\", \"Abstract Authors\", \"Abstract Affiliations\", \"Presentation Title (lowercase)\",\n","    \"Abstract\", \"Abstract Url\", \"Company\", \"Asset\", \"Indication\", \"MoA\", \"Other Keywords\"\n","]\n","\n","# Initialize the combined data list\n","combined_data = []\n","\n","\n","\n","# Iterate over each session session\n","for session in dataSession:\n","    # Common session-related data (outside presentation loop)\n","    session_common_data = {\n","        \"Session Date\": session.get('day', {}).get(\"date\"),\n","        \"Session Start Time\": session.get(\"start_time\"),\n","        \"Session End Time\": session.get(\"end_time\"),\n","        \"Session Timezone\": \"CEST\",\n","        \"Session Location\": session.get(\"room\", {}).get('short_name'),\n","        \"Session Number\": session.get(\"id\"),\n","        \"Session Pathways\": ', '.join([pathway.get('name') for pathway in session.get('pathways', [])]),\n","        \"Session CME\": \"\",\n","        \"Session Type\": session.get('session_type', {}).get('name'),\n","        \"Session Group\": session.get(\"session_group_id\"),\n","        \"Session Title\": clean_text(session.get(\"title\")),\n","        \"Session Description\": session.get(\"content\", {}).get(\"outline\"),\n","    }\n","\n","    for presentation in session.get('presentations', []):\n","        presentation_title = presentation.get(\"presentation\", {}).get(\"title\", \"\")\n","\n","        Abstract_url = None\n","        Abstract_id = presentation.get(\"presentation\", {}).get(\"abstract_id\")\n","        if (Abstract_id):\n","            Abstract_url = f\"https://programme.ueg.eu/week2024/#/details/presentations/{Abstract_id}\"\n","            Abstract_endpoint = f\"https://ueg2024.abstract.documedias.systems/api/v1/manager/abstract/multi/html/id/{Abstract_id}/template/planner_preview?program_type=week&system=program_week_2024\"\n","            responseAbstract = requests.get(Abstract_endpoint)\n","            dataAbstract = responseAbstract.json()\n","\n","            abstracts = dataAbstract.get(f\"{Abstract_id}\")\n","            print(abstracts)\n","\n","        authors = []\n","        institutions = []\n","\n","        for person in presentation.get('presentation', {}).get('persons', []):\n","            #print(person)\n","            institution = person.get('person', {}).get('institution')\n","            author_name = person.get('person', {}).get('first_name') + ' ' + person.get('person', {}).get('last_name')\n","            if institution:\n","                institutions.append(institution)\n","                #print(institutions)\n","            if author_name:\n","                authors.append(author_name)\n","\n","        presentation_title = presentation.get(\"presentation\", {}).get(\"title\", \"\")\n","        session_title = session.get(\"title\", \"\")\n","        session_description = session.get(\"content\", {}).get(\"outline\", \"\")\n","        \n","        # Find matching keywords in the titles, descriptions, and abstracts\n","        matched_indication = find_matching_keywords(presentation_title, indication) or \\\n","                             find_matching_keywords(session_title, indication) or \\\n","                             find_matching_keywords(session_description, indication)\n","\n","        matched_asset = find_matching_keywords(presentation_title, asset) or \\\n","                        find_matching_keywords(session_title, asset) or \\\n","                        find_matching_keywords(session_description, asset)\n","\n","        matched_company = find_matching_keywords(presentation_title, company_name) or \\\n","                          find_matching_keywords(session_title, company_name) or \\\n","                          find_matching_keywords(session_description, company_name)\n","\n","        matched_MoA = find_matching_keywords(presentation_title, MoA) or \\\n","                      find_matching_keywords(session_title, MoA) or \\\n","                      find_matching_keywords(session_description, MoA)\n","\n","        matched_other_keywords = find_matching_keywords(presentation_title, other_keywords) or \\\n","                                 find_matching_keywords(session_title, other_keywords) or \\\n","                                 find_matching_keywords(session_description, other_keywords)\n","\n","\n","        presentation_data = {\n","            **session_common_data,  # Include session data in each presentation entry\n","            \"Session Format\": ', '.join([format.get('name') for format in session.get('formats', [])]),\n","            \"Presentation Date\": session.get('day', {}).get(\"date\"),\n","            \"Presentation Location\": session.get(\"room\", {}).get('short_name'),\n","            \"Presentation Start Time\": presentation.get(\"start_time\"),\n","            \"Presentation End Time\": presentation.get(\"end_time\"),\n","            \"Presentation ID\": presentation.get(\"id\"),\n","            \"Presentation Number\": presentation.get(\"presentation\", {}).get(\"code\"),\n","            \"Presentation Title\": presentation.get(\"presentation\", {}).get(\"title\"),\n","            \"Presentation Title (lowercase)\": presentation.get(\"presentation\", {}).get(\"title\", '').lower(),\n","            \"Abstract Authors\":', '.join(authors),\n","            \"Abstract Affiliations\": ', '.join(institutions),  # Join institutions into a string\n","            \"Abstract\": abstracts,  # IF ABSTRACT_ID:\n","            \"Abstract Url\": Abstract_url,\n","            \"Company\": matched_company,\n","            \"Asset\": matched_asset,\n","            \"Indication\": matched_indication,\n","            \"MoA\": matched_MoA,\n","            \"Other Keywords\": matched_other_keywords,\n","        }\n","\n","\n","\n","        #print(author_name)\n","        print(combined_data)\n","        # Append combined data\n","        combined_data.append(presentation_data)"]},{"cell_type":"code","execution_count":13,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":308,"status":"ok","timestamp":1728940741006,"user":{"displayName":"Cyro Queiroz","userId":"10205326658499773298"},"user_tz":180},"id":"WsWafeKiX_rT","outputId":"89b1e526-3331-4ebe-c307-66f775a21f8e"},"outputs":[{"name":"stdout","output_type":"stream","text":["  Session Date Session Start Time Session End Time Session Timezone  \\\n","0   2024-10-14              08:30            09:30             CEST   \n","1   2024-10-14              08:30            09:30             CEST   \n","2   2024-10-14              08:30            09:30             CEST   \n","3   2024-10-14              08:30            09:30             CEST   \n","\n","  Session Location  Session Number Session Pathways        Session Format  \\\n","0               A3              80              IBD  Live, Recording, Q&A   \n","1               A3              80              IBD  Live, Recording, Q&A   \n","2               A3              80              IBD  Live, Recording, Q&A   \n","3               A3              80              IBD  Live, Recording, Q&A   \n","\n","  Session CME      Session Type  ...           Abstract Authors  \\\n","0              Abstract Session  ...             Bruce E. Sands   \n","1              Abstract Session  ...  Manuel Barreiro de Acosta   \n","2              Abstract Session  ...             David T. Rubin   \n","3              Abstract Session  ...         Tadakazu Hisamatsu   \n","\n","                               Abstract Affiliations  \\\n","0            Icahn School of Medicine at Mount Sinai   \n","1  Health Research Institute of Santiago de Compo...   \n","2                     University of Chicago Medicine   \n","3               Kyorin University School of Medicine   \n","\n","                      Presentation Title (lowercase)  \\\n","0  efficacy of guselkumab versus placebo in crohn...   \n","1  real-world short-term effectiveness of risanki...   \n","2  extended induction response over time in patie...   \n","3  efficacy and safety of guselkumab maintenance ...   \n","\n","                                            Abstract  \\\n","0  <i><u>B.E. Sands</u><sup>1</sup>, G. D’Haens<s...   \n","1  <i><u>M. Barreiro-de Acosta</u><sup>1,2</sup>,...   \n","2  <i><u>D.T. Rubin</u><sup>1</sup>, A. Charabaty...   \n","3  <i>D.T. Rubin<sup>1</sup>, A. Dignass<sup>2</s...   \n","\n","                                        Abstract Url Company         Asset  \\\n","0  https://programme.ueg.eu/week2024/#/details/pr...    None    Guselkumab   \n","1  https://programme.ueg.eu/week2024/#/details/pr...    None  Risankizumab   \n","2  https://programme.ueg.eu/week2024/#/details/pr...    None   Mirikizumab   \n","3  https://programme.ueg.eu/week2024/#/details/pr...    None    Guselkumab   \n","\n","                       Indication   MoA Other Keywords  \n","0                           Crohn  None           None  \n","1                           Crohn  None           None  \n","2  Ulcerative Colitis; Ulcerative  None           None  \n","3                             IBD  None           None  \n","\n","[4 rows x 30 columns]\n"]}],"source":["\n","df_combined = pd.DataFrame(combined_data, columns=fields)\n","df_combined.to_excel(\"UEG_Scrape_Combined.xlsx\", sheet_name=\"Combined Data\", index=False)\n","print(df_combined)"]}],"metadata":{"colab":{"authorship_tag":"ABX9TyM8UexKAcrIDLVHZuir5IVg","provenance":[]},"kernelspec":{"display_name":".venv","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.12.3"}},"nbformat":4,"nbformat_minor":0}
